presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java

```diff
@@ -14,14 +14,20 @@
 package com.facebook.presto.server;
 
 import com.google.common.base.StandardSystemProperty;
+import com.sun.management.UnixOperatingSystemMXBean;
 import io.airlift.slice.Slice;
 import io.airlift.slice.Slices;
 import org.joda.time.DateTime;
 
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
 import java.nio.ByteOrder;
 
 final class PrestoSystemRequirements
 {
+
+    private static final int MIN_FILE_DESCRIPTORS = 10_000;
```

Comment:
This should probably be a system property like `presto.min-file-descriptors`, just in case people need to override this (perhaps for testing).